pkg/services/preference/prefimpl/pref.go

```diff
@@ -170,6 +174,15 @@ func (s *Service) Patch(ctx context.Context, cmd *pref.PatchPreferenceCommand) e
 		preference.JSONData.Language = *cmd.Language
 	}
 
+	if cmd.Navbar != nil {
+		if preference.JSONData == nil {
+			preference.JSONData = &pref.PreferenceJSONData{}
+		}
```

Comment:
nit: I think that we could move this inside the below conditional (no reason to set `JSONData` if `cmd.Navbar.SavedItemIds == nil`)

Reply:
Thanks! Moved that inside!